### PR TITLE
feat: img

### PR DESCRIPTION
**Pull request by gavmck**
Image utility component

Gives props for:
- block
- responsive
- radius
- aspect ratio
- object fit
- object position

### CODE REVIEW

File: components/content/Img/Img.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      width: 100%;
+    }
+  }
+
+  &[radius] {
+    border-radius: var(--diamond-radius);
+    overflow: hidden;
+  }
+
+  &[aspect-ratio],
```
Comment (DanWebb) [inline components/content/Img/Img.css:26]: Why are we setting the aspect ratio of the container?

File: components/content/Img/Img.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+diamond-img {
+  display: inline-block;
+
+  &[block] {
```
Comment (DanWebb) [inline components/content/Img/Img.css:4]:
```
&[block],
&[block] img
```

File: components/content/Img/Img.stories.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { StoryObj } from '@storybook/web-components';
+import { html } from 'lit';
+
+import './Img';
+
+export default {
+  component: 'diamond-img',
+  argTypes: {
```
Comment (DanWebb) [inline components/content/Img/Img.stories.ts:8]: Should there be more argTypes here or are they guessed from Img.args?

File: components/content/Img/Img.ts [hunk 4]
```diff
@@ -1,50 +1,56 @@
...
-    ::slotted(img[fit='fill']) {
-      object-fit: fill;
-    }
-  `;
+  @property({ type: Boolean, reflect: true }) block = false;
+  @property({ type: Boolean, reflect: true }) responsive = false;
+  @property({ reflect: true, attribute: 'aspect-ratio' }) aspectRatio: string =
+    'auto';
+  @property({ type: Boolean, reflect: true }) radius = false;
+  @property({ reflect: true, attribute: 'object-fit' }) objectFit:
```
Comment (DanWebb) [inline components/content/Img/Img.ts:22]:
```suggestion
  @property({ reflect: true, attribute: 'object-fit' }) objectFit?:
```

File: components/content/Img/Img.ts [hunk 5]
```diff
@@ -1,50 +1,56 @@
...
+    | 'fill'
+    | 'cover'
+    | 'contain' = 'fill';
+  @property({ reflect: true, attribute: 'object-position' })
+  objectPosition: string = '50% 50%';
```
Comment (DanWebb) [inline components/content/Img/Img.ts:27]:
```suggestion
  objectPosition = '50% 50%';
```
TS will know it's a string

File: components/content/Img/Img.ts [hunk 6]
```diff
@@ -1,50 +1,56 @@
...
 
   render() {
-    return html`<slot></slot>`;
+    console.info(this.aspectRatio);
```
Comment (DanWebb) [inline components/content/Img/Img.ts:30]:
```suggestion
```

File: components/content/Img/Img.ts [hunk 7]
```diff
@@ -1,50 +1,56 @@
...
-      object-fit: cover;
-    }
-
-    ::slotted(img[fit='fill']) {
-      object-fit: fill;
-    }
-  `;
+  @property({ type: Boolean, reflect: true }) block = false;
+  @property({ type: Boolean, reflect: true }) responsive = false;
+  @property({ reflect: true, attribute: 'aspect-ratio' }) aspectRatio: string =
```
Comment (DanWebb) [inline components/content/Img/Img.ts:19]:
```suggestion
  @property({ reflect: true, attribute: 'aspect-ratio' }) aspectRatio?: string =
```

**Review by DanWebb — CHANGES_REQUESTED** [review]

File: components/content/Img/Img.stories.ts (hunk 3 above)
Comment (gavmck) [reply components/content/Img/Img.stories.ts:8]: Boolean args are inferred from initial args, aspect ratio and object position are free text so no need to define.

**Review by DanWebb — APPROVED** [review]

**Pull request merged**